### ISSUE TITLE
Message fields are not named correctly

### ISSUE THREAD

**Issue by evenh** [HubSpot/jackson-datatype-protobuf#48]
Hi, thanks for a useful library :-)

I don't know if this is a configuration problem, but I'm trying to integrate this library with Spring Boot 2.0.1 (with Jackson defaults) and I am registering the `ProtobufModule` (v. 0.9.9-jackson2.9-proto3) with Jackson. It seems like the default naming strategy (lower camel case) isn't being correctly applied, for some reason and results in keys like `configurationparameters` and `foobar` instead of `configurationParameters` and `fooBar`.

Is this a bug or something wrong with my setup? If I try switching to another naming strategy it works as expected.

**Comment by marianiandre** [comment]
Hi, I've faced the same issue. 

Debugging the library I've found that:

1. The PropertyNamingStrategy.LOWER_CAMEL_CASE strategy (the one you're willing to apply) is an instance of class PropertyNamingStrategy

2. PropertyNamingStrategyWrapper contructor performs this check: 
   if (delegate instanceof PropertyNamingStrategyBase) {
      this.delegate = (PropertyNamingStrategyBase) delegate;
    } else {
      this.delegate = SNAKE_TO_CAMEL;
    }
This means that, if you apply the LOWER_CAMEL_CASE, the strategy will be overwritten with SNAKE_TO_CAMEL because of a PropertyNamingStrategyBase base class check

Therefore, in order to have the correct java camel case naming I've implemented a dummy class:

public class JavaCamelCaseNamingStrategy extends PropertyNamingStrategyBase {
    @Override
    public String translate(String input)
    {
       return input;
    }
}

and set this class as naming strategy with mapper.setPropertyNamingStrategy(new JavaCamelCaseNamingStrategy())

Hope this will help. bye

**Comment by cowtowncoder** [comment]
@evenh apologies for missing this bug report. Whether it is a bug or not is easiest say seeing a test case of some kind.

@marianiandre Thank you for adding more information. One more question: is this with latest (2.9.6) version? (I assume so, just verifying).

I'd love to fix this problem, and at this point what would be of great help would be simple reproduction.
I can try creating one when I have time, but if anyone has something available that would be great.

**Comment by cowtowncoder** [comment]
(apologies for noise: added a note before realizing this was not about protobuf dataformat module)

**Comment by jhaber** [comment]
Sorry for missing this. Thanks for the investigation @marianiandre, your analysis appears correct to me. I think we can improve this experience by special-casing `PropertyNamingStrategy.LOWER_CAMEL_CASE` so that it behaves as expected out of the box (the other built-in naming strategies all appear to extend `PropertyNamingStrategyBase`).

**Comment by evenh** [comment]
I've created [a sample application](https://github.com/evenh/jackson-datatype-protobuf-48) to reproduce the issue, as well as demonstrating @marianiandre's workaround

**Comment by jhaber** [comment]
Thanks for that example. I just finished releasing version 0.9.10 which has a fix for using `PropertyNamingStrategy.LOWER_CAMEL_CASE`. I made a fork of that repo and pushed a commit: https://github.com/jhaber/jackson-datatype-protobuf-48/commit/c6fef9d8caec5fbb95709b164585d9cf13cb6538 showing how this would work in version 0.9.10. Unfortunately, it is still necessary to explicitly set `PropertyNamingStrategy.LOWER_CAMEL_CASE`, because by default we use this naming strategy: https://github.com/HubSpot/jackson-datatype-protobuf/blob/0e20be2bff6d6def4cf30a9bc0ea481b843fce7a/src/main/java/com/hubspot/jackson/datatype/protobuf/PropertyNamingStrategyWrapper.java#L29-L36

In your example app, the root issue is that the field names are camelCase instead of lowercase_with_underscores. This library assumes that the proto field names with be lowercase with underscores, which is recommended in the protobuf style guide: https://developers.google.com/protocol-buffers/docs/style#message-and-field-names
> Use underscore_separated_names for field names – for example, song_name.

I pushed another commit: https://github.com/jhaber/jackson-datatype-protobuf-48/commit/f18ac796440b9893df2e4580179a7750937d8064 which switches to underscore names, at which point the app works out of the box without messing with property naming strategies at all
